server/apps/immich/src/modules/immich-jwt/strategies/public-share.strategy.ts

```diff
@@ -0,0 +1,54 @@
+import { UserEntity } from '@app/database';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { InjectRepository } from '@nestjs/typeorm';
+import { ShareService } from 'apps/immich/src/api-v1/share/share.service';
+import { IStrategyOptions, Strategy } from 'passport-http-header-strategy';
+import { Repository } from 'typeorm';
+
+export const PUBLIC_SHARE_STRATEGY = 'public-share';
+
+const options: IStrategyOptions = {
+  header: 'x-immich-share-key',
+  param: 'key',
+};
+
+export class PublicUser extends UserEntity {
+  isPublicUser?: boolean;
+}
+
+@Injectable()
+export class PublicShareStrategy extends PassportStrategy(Strategy, PUBLIC_SHARE_STRATEGY) {
+  constructor(
+    private shareService: ShareService,
+    @InjectRepository(UserEntity)
+    private usersRepository: Repository<UserEntity>,
+  ) {
+    super(options);
+  }
+
+  async validate(key: string): Promise<PublicUser> {
+    const validatedLink = await this.shareService.getSharedLinkByKey(key);
+
+    if (validatedLink.expiresAt) {
+      const now = new Date().getTime();
+      const expiresAt = new Date(validatedLink.expiresAt).getTime();
+
+      if (now > expiresAt) {
+        throw new BadRequestException('Expired link');
+      }
+    }
+
+    const user = await this.usersRepository.findOne({ where: { id: validatedLink.userId } });
+
+    if (!user) {
+      throw new BadRequestException('Failure to validate public share payload');
+    }
+
+    let publicUser = new PublicUser();
+    publicUser = user;
+    publicUser.isPublicUser = true;
+
+    return publicUser;
```

Comment:
We should be using UserCore here to get by id, no querying the database directly. Also, we should change this to return `AuthUserDto`, which can include additional details like public = true and canUpload = true/false.

Reply:
If we use `UserCore` here, we have to import `IUserRepository` which is imported from `UserModule`, when I do that, there are lots of errors in the `ImmichJwtModule`

Reply:
Booooo. I guess we can defer this...